### PR TITLE
feat: retrieve catalog objects

### PR DESCRIPTION
**Pull request by BruceClark406**

### CODE REVIEW

File: lib/src/functions_model/batch_retrieve_catalog_objects.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  Map<String, dynamic> toJson() =>
+      _$BatchRetrieveCatalogObjectsRequestToJson(this);
+
+  final List<String> objectIds;
+  final bool? includeRelatedObjects;
+  final int? catalogVersion;
+  final bool? includeDeletedObjects;
+
+  @override
+  List<Object?> get props => [];
```
Comment (mtwichel) [inline lib/src/functions_model/batch_retrieve_catalog_objects.dart:32]: Add props?

File: lib/src/functions_model/retrieve_catalog_objects.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:square_connect/square_connect.dart';
+
+part 'retrieve_catalog_objects.g.dart';
+
+@JsonSerializable()
+class RetrieveCatalogObjectsResponse extends SquareResponse {
```
Comment (mtwichel) [inline lib/src/functions_model/retrieve_catalog_objects.dart:7]:
```suggestion
class RetrieveCatalogObjectResponse extends SquareResponse {
```
This is to retrieve one object right?

**Review by mtwichel — CHANGES_REQUESTED** [review]
Just a few nits 🙂

File: lib/src/functions_model/retrieve_catalog_objects.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  factory RetrieveCatalogObjectsResponse.fromJson(
+    Map<String, dynamic> json,
+  ) =>
+      _$RetrieveCatalogObjectsResponseFromJson(json);
+
+  /// Converts a [RetrieveCatalogObjectsResponse] to a [Map]
+  Map<String, dynamic> toJson() =>
+      _$RetrieveCatalogObjectsResponseToJson(this);
+
+  final List<CatalogObject> objects;
```
Comment (mtwichel) [inline lib/src/functions_model/retrieve_catalog_objects.dart:24]:
```suggestion
  final CatalogObject object;
```

**Review by mtwichel — CHANGES_REQUESTED** [review]

File: lib/src/functions_model/retrieve_catalog_objects.dart (hunk 2 above)
Comment (BruceClark406) [reply lib/src/functions_model/retrieve_catalog_objects.dart:7]: nice catch!

File: lib/src/functions_model/batch_retrieve_catalog_objects.dart (hunk 1 above)
Comment (BruceClark406) [reply lib/src/functions_model/batch_retrieve_catalog_objects.dart:32]: nice!

**Pull request merged**